app/Http/Controllers/Api/LocationsController.php

```diff
@@ -253,8 +253,12 @@ public function destroy($id)
      */
     public function selectlist(Request $request)
     {
-
-        $this->authorize('view.selectlists');
+        // If a user is in the process of editing their profile, as determined by the referrer,
```

Comment:
I think I'd rather see this in the Provider where we decide whether or not you can see the lists, versus pooping on the controller?

Reply:
I think of the `AuthServiceProvider` as purely registering gate definitions while controllers (in this case) use the registered gates. 

In this case, I think seeing code that checks http headers in the `AuthServiceProvider` would be confusing out of context while seeing it in the controller gives future developers the context right away. The controller is doing the work to say "check one of these gates depending on the context of the request" which I think is unique enough to live here.

Reply:
I disagree, but accept :)